cmake/Modules/FindPythonExtra.cmake

```diff
@@ -17,211 +17,84 @@
+#   If more components from FindPython3.cmake are needed, then find them manually before
+#   or after finding this module.
 #
 ###############################################################################
 
 # lint_cmake: -convention/filename, -package/stdargs
 
-set(PythonExtra_FOUND FALSE)
-
-# Prevent find_package(PythonLibs) from getting confused.
-unset(PYTHON_LIBRARY)
-
-find_package(PythonInterp 3.6 REQUIRED)
-
-if(PYTHONINTERP_FOUND)
-  if(APPLE)
-    find_program(PYTHON_CONFIG_EXECUTABLE NAMES "python3-config")
-    if(NOT PYTHON_CONFIG_EXECUTABLE)
-      message(FATAL_ERROR "Cannot find python3-config executable")
-    endif()
-
-    if(NOT DEFINED PythonExtra_INCLUDE_DIRS)
-      execute_process(
-        COMMAND
-        "${PYTHON_CONFIG_EXECUTABLE}"
-        "--includes"
-        OUTPUT_VARIABLE _output
-        RESULT_VARIABLE _result
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-      )
-      if(NOT _result EQUAL 0)
-        message(FATAL_ERROR
-          "execute_process(${PYTHON_CONFIG_EXECUTABLE} --includes) returned "
-          "error code ${_result}")
-      endif()
-
-      string(REPLACE " " ";" _output_list ${_output})
-
-      foreach(_includedir ${_output_list})
-        string(SUBSTRING "${_includedir}" 2 -1 _includedir)
-        list(APPEND PythonExtra_INCLUDE_DIRS "${_includedir}")
-      endforeach()
-    endif()
-    set(PythonExtra_INCLUDE_DIRS
-        ${PythonExtra_INCLUDE_DIRS}
-        CACHE INTERNAL
-        "The paths to the Python include directories.")
-    message(STATUS "Using PythonExtra_INCLUDE_DIRS: ${PythonExtra_INCLUDE_DIRS}")
-
-    if(NOT DEFINED PythonExtra_LIBRARIES)
-      execute_process(
-        COMMAND
-        "${PYTHON_CONFIG_EXECUTABLE}"
-        "--ldflags"
-        OUTPUT_VARIABLE _output
-        RESULT_VARIABLE _result
-        OUTPUT_STRIP_TRAILING_WHITESPACE
-      )
-      if(NOT _result EQUAL 0)
-        message(FATAL_ERROR
-          "execute_process(${PYTHON_CONFIG_EXECUTABLE} --ldflags) returned "
-          "error code ${_result}")
-      endif()
-
-      string(REPLACE " " ";" _output_list "${_output}")
-      set(PythonExtra_LIBRARIES
-        ""
-        CACHE INTERNAL
-        "The libraries that need to be linked against for Python extensions.")
+if(PythonExtra_FOUND)
+  return()
+endif()
 
-      set(_library_paths "")
-      foreach(_item ${_output_list})
-        string(REGEX MATCH "-L(.*)" _regex_match ${_item})
-        if(NOT _regex_match STREQUAL "")
-          string(SUBSTRING "${_regex_match}" 2 -1 _library_path)
-          list(APPEND _library_paths "${_library_path}")
-        endif()
-      endforeach()
+find_package(Python3 REQUIRED COMPONENTS Interpreter)
 
-      set(_python_version_no_dots
-        "${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR}")
-      set(_python_version
-        "${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}")
+get_target_property(PYTHON_EXECUTABLE Python3::Interpreter LOCATION)
```

Comment:
Since the line above did `find_package(Python3 REQUIRED COMPONENTS Interpreter)` The correct variable here is `Python3_EXECUTABLE`, not `PYTHON_EXECUTABLE`.

Reply:
Oh wait, my comment makes no sense. We're creating the variable `PYTHON_EXECUTABLE` 🤦 .